evalml/pipelines/pipeline_base.py

```diff
@@ -425,6 +426,60 @@ def feature_importance(self):
         df = pd.DataFrame(importance, columns=["feature", "importance"])
         return df
 
+    def graph_json(self):
+        """Generates a JSON with nodes consisting of the component names and parameters, and edges detailing component relationships.
+
+        Returns:
+            dag_json (str): A serialized JSON representation of a DAG structure.
+        """
+        dag_json = {"Nodes": {"X": "X", "y": "y"}, "Edges": {}}
+
+        for (
+            component_name,
+            component_info,
+        ) in self.component_graph.component_dict.items():
+            dag_json["Nodes"][component_name] = {
+                "class": component_info[0]
+                if isinstance(component_info[0], str)
+                else component_info[0].name,
+                "attributes": self.parameters[component_name]
+                if component_name in self.parameters
+                else None,
+            }
+
+        for (
+            component_name,
+            component_info,
+        ) in self.component_graph.component_dict.items():
+            for input_ in component_info[1:]:
+                from_comp = input_[:-2]
+                if input_.endswith(".x"):
+                    dag_json["Edges"][f"from_{from_comp}_to_{component_name}"] = {
+                        "from": from_comp,
+                        "to": component_name,
+                        "data": f"X modified by {from_comp}",
```

Comment:
`data` is meant to provide a little more detail as to what's being passed through that edge even though it can be inferred based on the `from` and `to` fields. OS users might find this useful in understanding this output.